scripts/main.js

```diff
@@ -0,0 +1,191 @@
+    row.setAttribute("key", itemObj.index);
+    titleCell.id = "title-text"
+    statusCell.innerHTML = itemObj.status ? checkIcon: "";
+    titleCell.innerHTML = itemObj.title;
+    dueDateCell.innerHTML = itemObj.date;
+    editTitleModeCell.value = itemObj.title;
+    deleteCell.textContent = deleteIcon;
+    editCell.textContent = editIcon
+    statusCell.className = "hover-cell"
+    editCell.className = "hover-cell"
+    deleteCell.className = "hover-cell"
+
+
+
+    titleEditToggleCell.appendChild(editTitleModeCell);
+    titleEditToggleCell.appendChild(titleCell);
+  
+    titleEdditButtonsCell.appendChild(deleteCell);
+    titleEdditButtonsCell.appendChild(editCell);
+    titleEdditButtonsCell.appendChild(titleEditToggleCell);
+
+    dueDateCell.addEventListener("dragstart", (event) => startDrag(event));
+    dueDateCell.addEventListener("dragover", (event) => dragOver(event));
+
+
+    deleteCell.addEventListener("click", (event) =>
+        removeItem(event.target.parentElement.parentElement));
+
+    statusCell.addEventListener("click", (event) =>
+        updateStatus(event.target.parentElement));
+
+    editCell.addEventListener("click", (event) =>
+toggleEditMode(event.target.parentElement));
+
+    editTitleModeCell.addEventListener("keypress", (event) => {
+        if (event.key === "Enter") {
+            event.preventDefault();
+            const newTitle = event.target.value;
+            const row = event.target.parentElement.parentElement.parentElement;
+            const titleNode = event.target.parentElement;
+            updateTitle(row, titleNode, newTitle);
+
+        }
+    });
+
+
+}
+/**
+ * 
+ * @param {*} row - row in table HTML element 
+ * @param {*} titleNode - title of task HTML element
+ * @param {*} newTitle - new title to replace old one
+ */
+function updateTitle(row, titleNode, newTitle) {
+    const key = row.getAttribute("key");
+    const itemToUpdate = JSON.parse(localStorage.getItem(key));
+    localStorage.removeItem(key);
+    itemToUpdate.title = newTitle;
+    localStorage.setItem(key, JSON.stringify(itemToUpdate));
+    titleNode.lastChild.innerHTML = newTitle;
+    toggleEditMode(row.firstChild); // change back to display mode
+
+}
+
+
+function createNewItem(title, date) {
+    let length = Number(localStorage.getItem("length"));
+    const newItem = {
+        title: title,
+        date: date,
+        status: false,
+        index: length++,
+    }
+    localStorage.setItem("length", length);
+    localStorage.setItem(newItem.index.toString(), JSON.stringify(newItem))
+    renderItem(newItem);
+
+}
+
+function removeItem(row) {
+    const key = row.getAttribute("key")
+    let length = localStorage.getItem("length");
+    let next = row
+    localStorage.removeItem(key);
+    for (let i = Number(key) + 1; i < length; i++) { // undate succesive rows
+        next = next.nextElementSibling;
+        updateItemIndex(i, next);
+    }
+    length--;
+    localStorage.setItem("length", length);
```

Comment:
you save each object in the local storage in its own key, maybe it will be easier for you to save 1 entry which is an array of objects?

Reply:
just thought that loading and downloading the entire todo list for every individual read/write would be less efficient.